error/s2n_errno.h

```diff
@@ -141,6 +141,8 @@ typedef enum {
     S2N_ERR_INVALID_SCT_LIST,
     S2N_ERR_INVALID_OCSP_RESPONSE,
     S2N_ERR_CANCELLED,
+    S2N_ERR_INVALID_MAX_FRAG_LEN,
```

Comment:
Our error "category" design is not very clear. You should add errors in this enum according the category that they are associated with. The comments delimit the start of each error type. For these two errors you added, they make the most sense to the in the `S2N_ERR_T_PROTO` section since they are errors encountered while processing data from the peer.